src/app/clusters/ethernet_network_diagnostics_server/ethernet_network_diagnostics_server.cpp

```diff
@@ -16,11 +16,106 @@
  */
 
 #include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterceptor.h>
 #include <app/CommandHandler.h>
+#include <app/MessageDef/AttributeDataElement.h>
 #include <app/util/af.h>
+#include <app/util/attribute-storage.h>
+#include <lib/core/Optional.h>
+#include <platform/ConnectivityManager.h>
 
 using namespace chip;
+using namespace chip::app;
 using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::EthernetNetworkDiagnostics::Attributes;
+using chip::DeviceLayer::ConnectivityManager;
+
+namespace {
+
+class EthernetDiagosticsAttrInterceptor : public AttributeAccessInterceptor
+{
+public:
+    // Register for the EthernetNetworkDiagnostics cluster on all endpoints.
+    EthernetDiagosticsAttrInterceptor() :
+        AttributeAccessInterceptor(Optional<EndpointId>::Missing(), EthernetNetworkDiagnostics::Id)
+    {}
+
+    CHIP_ERROR ReadAttributeData(ClusterInfo & aClusterInfo, TLV::TLVWriter * aWriter, bool * aDataRead) override;
+
+    bool IsRegistered() const { return sIsRegistered; }
+
+    void RegisterIfNeeded()
+    {
+        if (!sIsRegistered)
+        {
+            registerAttributeAccessInterceptor(this);
+        }
+        sIsRegistered = true;
```

Comment:
I don't see any logic to un-register this status, it this by design?

Reply:
Yes.  There is no need to un-register this, because it's not tied to a specific endpoint.

Reply:
Is there a motivating reason why we would need to de-register this?